caffe2/CMakeLists.txt

```diff
@@ -1125,6 +1125,36 @@ elseif(USE_CUDA)
 endif()
 
 if(USE_XPU)
+  function(clone_and_checkout REPO_URL DEST_DIR COMMIT_FILE)
```

Comment:
Cmake has native support for fetching git now since CMake 3.14

'''
include(FetchContent)

FetchContent_Declare(
  some_lib
  GIT_REPOSITORY https://github.com/.../repo.git
  GIT_TAG        v1.2.3
  GIT_SHALLOW    TRUE
)
FetchContent_MakeAvailable(some_lib)
'''

Reply:
Hi, @Skylion007  @EikanWang , the `FetchContent_Declare` and `FetchContent_MakeAvailable` includes clone, checkout and add_subdirectory. For torc-xpu-ops, it need to specify the output binary directory when add_subdirectory: 
``` 
add_subdirectory(${TORCH_ROOT}/third_party/torch-xpu-ops
      ${CMAKE_BINARY_DIR}/caffe2/aten_xpu)
```
But `FetchContent_Declare` does not support it.
And for torch-xpu-tla (intel cutlass) we only want to the header files in the repo, no need to build it, so add_subdirectory is not need.

Base on that, I think current implementation is more flexible.

Reply:
@etaf , I cannot capture your point. Do you mean even `FetchContent_Populate` does not work?

Reply:
Hi, @EikanWang : 
  Calling FetchContent_Populate(torch-xpu-tla) is deprecated, call
  FetchContent_MakeAvailable(torch-xpu-tla-repo) instead.  Policy CMP0169 can
  be set to OLD to allow FetchContent_Populate(torch-xpu-tla-repo) to be
  called directly for now, but the ability to call it with declared details
  will be removed completely in a future version.

Reply:
Hmm... Thanks for the information @etaf . @Skylion007 any comments?